packages/server/lib/browsers/cdp_automation.ts

```diff
@@ -116,6 +116,11 @@ export const CdpAutomation = (sendDebuggerCommandFn: SendDebuggerCommand) => {
       }
     }
 
+    if (setCookieRequest.name.startsWith('__Host-')) {
+      setCookieRequest.url = `https://${cookie.domain}`
+      delete setCookieRequest.domain
+    }
```

Comment:
is this part of an RFC somewhere? i'm not familiar with the `__Host-` cookies

if it's just that __Host- cookies need to be set on secure domains only, would this work instead?

```suggestion
    if (setCookieRequest.name.startsWith('__Host-')) {
      setCookieRequest.secure = true
    }
```

Reply:
no, chrome will error unless we add `url` and remove `domain`. I found info on the __Host- prefix here https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

Reply:
interestingly if you use the extension cookie.set API, chrome will hang/never resolve when setting a __Host- cookie w/ a domain

Reply:
hmm, so is this going to "fix" broken usages of `__Host-` cookies?

what i mean is, if a user sets a `__Host-` cookie on an `http://` URL, it will fail to set in regular Chrome. but with this change, in Cypress, it *will* be set, just on the `https://` URL?

i'm wondering if the actual fix here is just to ignore errors setting cookies, since they typically represent invalid cookie usages...

Reply:
you make a good point. I think we should throw if secure wasn't specified in this case. The reason this needs special logic is that we don't expose the `cookie.url` option in the cy.setCookie api, and that is the **only** way to set a __Host- cookie

Reply:
ah i see, they're using `cy.setCookie`. yeah, it should probably throw if they explicitly set it, and be swallowed silently if it's during a cy.visit or normal HTTP request